script/docker_dev.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# Script to publish a docker container from the current state of the repo.
+# Useful for iterating on AWS
+rm dist/*.whl
+hatch build -t wheel
+python script/gen_req.py dist/*.whl > ./dist/_reqs.txt
```

Comment:
Okay, so I found that you can just use [`pip-tools`](https://pip-tools.readthedocs.io) rather than your own package to ensure that you get the full dependencies by reading in from `pyproject.toml`, leaving your wheel file alone.

Seems like there's a whole discussions (https://github.com/pypa/pip/issues/11584, ) on being able to pip install only dependencies, especially from a `pyproject.toml`. Someone even have the same exact use case as what you're doing here (https://github.com/pypa/pip/issues/11440).

```suggestion
pip install pip-tools # install pip tools
pip-compile --output-file=./dist/_reqs.txt pyproject.toml # export deps to reqs file
```

Reply:
Good to know. Unfortunately, its does the full dependency resolution, so it takes ~10 secs vs 4ms for the python script.

Reply:
Is the 4ms includes the docker build step as well? If that's the case, then your current way is probably good. I figured if you get the resolution ahead of time, then the docker build will be a lot faster since pip wouldn't have to resolve then in https://github.com/mdenolle/NoisePy/pull/183/files#diff-86930c95a19b82f7e64a962a0053d44e855824813019b3698eae4917a90cdcacR11